### PR TITLE
Set up server side

### PR DESCRIPTION
**Pull request by miyabitanimchi**
Set up the server side. Please let me know if there is anything we need to change before the actual development 🙏

### CODE REVIEW

File: server/controllers/signUp.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import User from "../models/userModels.js";
+import bcrypt from "bcrypt";
+
+export const createUser = async (req, res) => {
+  const { email, password } = req.body;
+  const saltPassword = await bcrypt.genSalt(10);
+  const securePassword = await bcrypt.hash(password, saltPassword);
+
+  const signedUser = new User({
+    email: email,
```
Comment (shiv-chan) [inline server/controllers/signUp.js:10]: Just `email` also works since they are the same name.

File: server/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+// set them to properly send a reqest
+app.use(bodyParser.json({ limits: "30mb", extended: true }));
+app.use(bodyParser.urlencoded({ limits: "30mb", extended: true }));
+app.use(cors());
+app.use("/login", loginRoute);
+app.use("/signup", signUpRoute);
+
+const DATABASE_URL = process.env.DATABASE_URL;
+const PORT = process.env.PORT || 5000; // for Heroku?
```
Comment (shiv-chan) [inline server/index.js:20]: `process.env.PORT` is for Heroku or any other deploy platform.  I believe wee are going to set PORT number in the platform's environment key setting.
For now, we are going to use post 5000 locally.

File: server/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import bodyParser from "body-parser";
+import loginRoute from "./routes/logIn.js";
+import signUpRoute from "./routes/signUp.js";
+
+const app = express();
+dotenv.config();
+
+// set them to properly send a reqest
+app.use(bodyParser.json({ limits: "30mb", extended: true }));
+app.use(bodyParser.urlencoded({ limits: "30mb", extended: true }));
```
Comment (shiv-chan) [inline server/index.js:14]: don't need `bodyParser` anymore. https://expressjs.com/en/api.html
`express.json()` can perse incoming request as itself in Express v4.16.0 onwards.

@miyabitanimchi did you get an error without it?

File: server/package.json [hunk 4]
```diff
@@ -11,6 +12,7 @@
   "description": "",
   "dependencies": {
     "bcrypt": "^5.0.1",
+    "body-parser": "^1.19.0",
```
Comment (shiv-chan) [inline server/package.json:15]: See this comment: https://github.com/shiv-chan/senpai-kohai/pull/2/files#r744211693

**Review by shiv-chan — APPROVED** [review]

**Comment by shiv-chan** [comment]
Approved👍

**Review by Dannyboy222 — APPROVED** [review]

**Pull request merged**